app/controllers/articles_controller.rb

```diff
@@ -5,35 +5,34 @@ class ArticlesController < ApplicationController
   def index
     if params[:section_id]
       @section = Section.friendly.find(params[:section_id])
-      @articles = Article.where("section_id = ?", @section.id)
+      @articles = Article
+                    .where("section_id = ?", @section.id)
+                    .joins("JOIN sections ON articles.id = sections.id")
+                    .order("articles.rank + 3 * sections.rank")
     else
-      @articles = Article.all
-    end
-    if params[:order_by] == 'rank'
-      @articles = @articles.order {|article| find_combined_rank(article)}.reverse
-    end
-    if params[:order_by] == 'date'
-      @articles = @articles.reverse
-    end
-    if params[:limit]
-      limit = params[:limit]
-      @articles = @articles.first(limit)
-    end
-    if params[:content] == 'false'
-      @articles = @articles.select(
-        :id,
-        :title,
-        :slug,
-        :volume,
-        :issue,
-        :is_published,
-        :created_at,
-        :updated_at,
-        :section_id,
-        :rank,
-        :summary
-      )
+      @articles = Article
+                   .joins("JOIN sections ON articles.id = sections.id")
+                   .order("articles.rank + 3 * sections.rank")
     end
+
+    @articles = @articles.order(:created_at).reverse if params[:order_by] == 'date'
```

Comment:
@NicholasLYang I added a condition here.

Reply:
Okay? Rewrite it?